### ISSUE TITLE
combine spacers and matchers

### ISSUE THREAD

**Issue by j4k0xb** [codemod-js/codemod#918]
I've been using this awesome package extensively in the past weeks and it supported almost every use-case, but ran into problems when matching a list with a variable number of values:
```js
React.createElement(type, props, ...children)
```
This is currently possible with `m.anyList(m.anyString(), m.objectExpression(), m.zeroOrMore())`, but the spacer matches any value without an easy way to filter it (especially with multiple distributed spacers).

So I propose adding an optional matcher parameter for all spacer functions:
```ts
const matcher = m.anyList(m.anyString(), m.oneOrMore(m.anyNumber()), m.anyString());
matcher.match(['', 1, 1, '']); // true
matcher.match(['', 1, null, '']); // false
```

Would like to hear your opinion on this

**Comment by eventualbuddha** [comment]
@j4k0xb thanks for the suggestion! I thought it was a good idea, so I implemented it in #920. Take a look and see if it works for you. I added your specific example as a test. If you try it out and it does what you need I'll merge it and do a release.

**Comment by j4k0xb** [comment]
Works perfectly, thanks for the quick implementation 👍🏻

**Comment by eventualbuddha** [comment]
Published in `@codemod/matchers@1.7.0`.
